mzLib/Readers/ExternalResults/BaseClasses/ResultFile.cs

```diff
@@ -12,14 +12,26 @@ public abstract class ResultFile<TResult> : IResultFile, IEquatable<ResultFile<T
         #region Base Properties
         public string FilePath { get; set; }
 
-        private List<TResult> _results;
+        private List<TResult>? _results;
         public List<TResult> Results
         {
             get
             {
-                if (!_results.Any())
+                // if the results have already been read in or set
+                if (_results is not null) 
+                    return _results;
+
+                // if the results have not been read, and the file exists, read the results
+                if (File.Exists(FilePath)) 
```

Comment:
It seems like just adding this line would be enough?

Reply:
Like, if (!_results.Any() & File.Exists(FilePath) ...
That would accomplish the same thing with fewer changes

Reply:
This is because of a specific case. The file exists, but there is nothing besides the header. In that case, the file will exist, the _results will be empty, and the LoadResults() method would be called, leading to the same crash.